proof/refine/ARM/Invariants_H.thy

```diff
@@ -997,6 +997,12 @@ definition "vs_valid_duplicates' \<equiv> \<lambda>h.
               x && ~~ mask (vs_ptr_align ko) = y && ~~ mask (vs_ptr_align ko) \<longrightarrow>
               h x = h y"
 
+definition valid_replies' :: "kernel_state \<Rightarrow> bool" where
```

Comment:
This name already exists in AInvs, actually. But maybe the one in AInvs could do a better name...hmm.

Reply:
And I am wondering how this relates to `valid_replies` in AInvs. I would hope crossing from abstract to concrete to hold. Crossing `sc_with_reply` will give you the `replyPrev/replyNext` assumption, I think, so probably it will work.